backend/cn/lib/core_to_mucore.ml

```diff
@@ -32,7 +32,7 @@ module SymSet = Set.Make (Sym)
 module SymMap = Map.Make (Sym)
 module IdMap = Map.Make (Id)
 open Print
-open Cn
+open CF.Cn
```

Comment:
It's possible to get rid of the open altogether (though that may be orthogonal to this PR).
```diff
12:51 ➜  cerberus git:(c76fe3d9f) ✗ git --no-pager diff --
diff --git a/backend/cn/lib/core_to_mucore.ml b/backend/cn/lib/core_to_mucore.ml
index 7f406095c..8284db9dc 100644
--- a/backend/cn/lib/core_to_mucore.ml
+++ b/backend/cn/lib/core_to_mucore.ml
@@ -32,7 +32,6 @@ module SymSet = Set.Make (Sym)
 module SymMap = Map.Make (Sym)
 module IdMap = Map.Make (Id)
 open Print
-open CF.Cn
 open Resultat

 open Effectful.Make (Resultat)
@@ -1238,7 +1237,7 @@ let add_spec_arg_renames
   loc
   args
   arg_cts
-  (spec : (Symbol.sym, Ctype.ctype) cn_fun_spec)
+  (spec : (Symbol.sym, Ctype.ctype) Cn.cn_fun_spec)
   env
   =
   List.fold_right
@@ -1248,7 +1247,7 @@ let add_spec_arg_renames
         fun_sym
         (Memory.sbt_of_sct (convert_ct loc ct))
         env)
-    (List.combine args (List.combine arg_cts spec.cn_spec_args))
+    (List.combine args (List.combine arg_cts spec.Cn.cn_spec_args))
     env


@@ -1318,13 +1317,13 @@ let normalise_fun_map_decl
                         ^^ P.colon
                         ^^^ !^"re-specification of CN annotations from:"
                         ^^ P.break 1
-                        ^^^ Locations.pp spec.cn_spec_loc)
+                        ^^^ Locations.pp spec.Cn.cn_spec_loc)
                  }
            in
            let env = add_spec_arg_renames loc args (List.map snd arg_cts) spec env in
            let env =
              C.add_renamed_computational
-               spec.cn_spec_ret_name
+               spec.Cn.cn_spec_ret_name
                ret_s
                (Memory.sbt_of_sct (convert_ct loc ret_ct))
                env
@@ -1382,7 +1381,7 @@ let normalise_fun_map_decl
        return (Some (M_Proc (loc, args_and_body, trusted, desugared_spec), mk_functions))
      | Mi_ProcDecl (loc, ret_bt, _bts) ->
        (match SymMap.find_opt fname fun_specs with
-        | Some (_ail_marker, (spec : (Symbol.sym, Ctype.ctype) cn_fun_spec)) ->
+        | Some (_ail_marker, (spec : (Symbol.sym, Ctype.ctype) Cn.cn_fun_spec)) ->
           let@ () =
             check_against_core_bt loc ret_bt (Memory.bt_of_sct (convert_ct loc ret_ct))
           in
@@ -1554,7 +1553,7 @@ let register_glob env (sym, glob) =

 (* |> C.add_c_var_value sym (IT.sym_ (sym, bt)) *)

-let translate_datatype env { cn_dt_loc; cn_dt_name; cn_dt_cases; cn_dt_magic_loc = _ } =
+let translate_datatype env Cn.{ cn_dt_loc; cn_dt_name; cn_dt_cases; cn_dt_magic_loc = _ } =
   let translate_arg (id, bt) =
     (id, SBT.to_basetype (Compile.translate_cn_base_type env bt))
   in
@@ -1585,7 +1584,7 @@ let normalise_file ~inherit_loc ((fin_markers_env : CAE.fin_markers_env), ail_pr
   let env = List.fold_left register_glob env globs in
   let fun_specs_map =
     List.fold_right
-      (fun (id, spec) acc -> SymMap.add spec.cn_spec_name (id, spec) acc)
+      (fun (id, spec) acc -> SymMap.add spec.Cn.cn_spec_name (id, spec) acc)
       ail_prog.cn_fun_specs
       SymMap.empty
   in
```

Reply:
In general, I certainly prefer fewer `open`s and more qualified names. My only concern would be someone seeing `Cn.x` here and thinking it was something local, rather than something from `Cerb_frontend`, but it isn't a strong concern. Happy to make the change if you like.

Reply:
Or perhaps I ought to defer to #433?

Reply:
Yeah, I did the diff and then got carried away... It's all sorted now. The librarification won't cause any issues with it since the name `Cn` will be unambiguous to the build system.